main/main.c

```diff
@@ -317,23 +317,85 @@ static PHP_INI_MH(OnSetSerializePrecision)
 static PHP_INI_MH(OnChangeMemoryLimit)
 {
 	size_t value;
-	if (new_value) {
-		value = zend_ini_parse_uquantity_warn(new_value, entry->name);
-	} else {
-		value = Z_L(1)<<30;		/* effectively, no limit */
-	}
-	if (zend_set_memory_limit(value) == FAILURE) {
-		/* When the memory limit is reset to the original level during deactivation, we may be
-		 * using more memory than the original limit while shutdown is still in progress.
-		 * Ignore a failure for now, and set the memory limit when the memory manager has been
-		 * shut down and the minimal amount of memory is used. */
-		if (stage != ZEND_INI_STAGE_DEACTIVATE) {
-			zend_error(E_WARNING, "Failed to set memory limit to %zd bytes (Current memory usage is %zd bytes)", value, zend_memory_usage(true));
+
+    if (new_value) {
```

Comment:
Please keep using tabs in C files.

Reply:
Oops - I'll be adding these back as well :)